lib/sms/sms.c

```diff
@@ -79,16 +80,28 @@ AT_MONITOR_ISR(sms_at_handler_cmt, "+CMT", sms_at_cmd_handler_cmt, PAUSED);
 AT_MONITOR_ISR(sms_at_handler_cds, "+CDS", sms_at_cmd_handler_cds, PAUSED);
 AT_MONITOR_ISR(sms_at_handler_cms, "+CMS", sms_at_cmd_handler_cms, PAUSED);
 
+static void sms_ack_resp_handler(const char *resp)
+{
+	/* No need to do anything really */
+	if (strlen(resp) < 2 ||
+	    (strlen(resp) >= 2 && resp[0] != 'O' && resp[1] != 'K')) {
```

Comment:
The second `strlen()` is redundant.
```suggestion
	if (strlen(resp) < 2 || (resp[0] != 'O' && resp[1] != 'K')) {
```